mantidproject/mantid
# Fitting tool in workbench

**Description of work.**
Added FitPropertyBrowser to plots as a docked widget. 

**To test:**
1. Open  a 1D plot.
2. Click the `Fit` tool button.
3. Check that a FitPropertyBrowser has showed up.
4. The plot must have two green vertical lines that mark the limits of the fitting region.
5. Move the lines with the mouse. This should change the StartX and EndX values.
6. Set a function. Run a fit.
7. Check that the calculated and the residuals lines are shown.

Fixes #24027



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
